master/internal/db/setup.go

```diff
@@ -49,3 +52,47 @@ func Setup(opts *config.DBConfig) (*PgDB, error) {
 	}
 	return db, nil
 }
+
+// ValidateRPWorkspaceBindings checks if rp-workspace bindings pertain to valid resource pools.
+// Bindings with resource pools that don't exist have the "valid" column set to false.
+func ValidateRPWorkspaceBindings(pools []config.ResourcePoolConfig) error {
+	var poolNames []string
+	for _, pool := range pools {
+		poolNames = append(poolNames, pool.PoolName)
+	}
+
+	ctx := context.Background()
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %t", err)
+	}
+	defer func() {
+		txErr := tx.Rollback()
+		if txErr != nil && txErr != sql.ErrTxDone {
+			log.WithError(txErr).Error("error rolling back transaction in AddExperiment")
```

Comment:
Just for my understanding, how is ValidateRPWorkspaceBindings() related to AddExperiment?

Reply:
It's not related to AddExperiment - we just run this on startup to validate resource pools